File: app/src/main/java/net/janczar/powertape/example/view/main/MainPresenter.java
package net.janczar.powertape.example.view.main;

import net.janczar.powertape.annotation.Inject;
import net.janczar.powertape.annotation.Provide;
import net.janczar.powertape.example.domain.repository.MessagesRepository;
import net.janczar.powertape.example.domain.usecase.GetMessagesUseCase;
import net.janczar.powertape.example.view.base.BasePresenter;

import java.util.List;

public class MainPresenter extends BasePresenter<MainView> {

    @Inject
    GetMessagesUseCase getMessagesUseCase;

    @Provide
    public MainPresenter() {
    }

    public void loadMessages() {
        manage(
            getMessagesUseCase.execute(null)
                .subscribe(
                        messages -> {
                            view.showMessages(messages);
                        }
                ));

    }
}

File: app/src/main/java/net/janczar/powertape/example/view/main/MainPresenter.kt
package net.janczar.powertape.example.view.main

import net.janczar.powertape.annotation.Inject
import net.janczar.powertape.annotation.InjectProperty
import net.janczar.powertape.annotation.Provide
import net.janczar.powertape.example.domain.repository.UserRepository
import net.janczar.powertape.example.domain.usecase.GetMessagesUseCase
import net.janczar.powertape.example.domain.usecase.GetUserUseCase
import net.janczar.powertape.example.view.base.BasePresenter
import net.janczar.powertape.log.Log


class MainPresenter @Provide constructor(
        private val getMessagesUseCase: GetMessagesUseCase
        //,private val getUserUseCase: GetUserUseCase
) : BasePresenter<MainView>() {

    @InjectProperty
    private val test: JustForTest by inject()

    fun loadMessages() {
        Log.i("DEBUG", "PRESENTER TEST: "+test.text)
        manage(
                getMessagesUseCase
                        .execute()
                        .subscribe { messages -> view?.showMessages(messages) }
        )
    }

}